doc/networking.md

```diff
@@ -107,22 +108,56 @@ The composition of the body (and certain headers) varies based on the content-ty
 Processing a `/segment` request consists of the following steps:
 
 1. Verify the credentials generated by the orchestrator
-2. Verify that the job hasn't expired.
-2. Verify the segment signature from the broadcaster.
-3. Download the body
-4. Verify the keccak256 hash of the body matches `SegData.hash`
-5. Transcode the segment
-6. Return 200 if transcoding success, non-200 otherwise.
+2. Verify that the job hasn't expired and is still claimable.
+3. Verify the segment signature from the broadcaster.
```

Comment:
Should we add a step for checking if the broadcaster's deposit is too low?

Reply:
Hmm. The deposit check is actually done outside the RPC/networking code, along with checking receipts for duplicate sequence numbers [1]. So far the docs have been confined to the code in `rpc.go` but should we expand that?

I'm a little hesitant to enumerate every possible failure condition since that makes it easier for the documentation to fall out of date. Perhaps more detailed docs are a good thing, even if it's more work to keep them up-to-date?

Along with @darkdarkdragon 's note about the `TranscoderBusy` error, this also brings up the question of what a *networking* spec should contain, and whether details on claimability or transcoder policy is appropriate for such a spec.

[1] On a related note, this presents an opportunity to refactor and consolidate the various checks into one function, but we need to be a little careful here. The claimable checks are done prior to download, while the receipt/deposit checks are done after download. Checking the sequence numbers prior to download gives us slightly weaker assurances since it widens the vulnerable gap to include in-flight segments.  So in that case we might actually want to do the checks twice -- once before download (during the `verifyToken check`), and once before transcoding, where the deposit/receipt checks are currently done.

Reply:
Good point on keeping these particular docs networking focused - maybe details like checking a broadcaster's deposit and checking for duplicate sequence numbers can be considered as a part of an orchestrator/transcoder's internal validation process. The networking spec would indicate that the orchestrator/transcoder's internal validation process occurs at some point, but wouldn't detail what the process actually consists of with the details being documented elsewhere (for now in comments)?

Reply:
Reworded step 7 from "Transcode the segment" to "Transcoder performs additional checks and transcodes the segment."